sql/executor.go

```diff
@@ -165,28 +165,47 @@ func (e Executor) execStmt(stmt parser.Statement, params parameters, planMaker *
 				} else {
 					switch vt := val.(type) {
 					case parser.DBool:
-						row.Values = append(row.Values, driver.Datum{BoolVal: (*bool)(&vt)})
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_BoolVal{BoolVal: bool(vt)},
+						})
 					case parser.DInt:
-						row.Values = append(row.Values, driver.Datum{IntVal: (*int64)(&vt)})
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_IntVal{IntVal: int64(vt)},
+						})
 					case parser.DFloat:
-						row.Values = append(row.Values, driver.Datum{FloatVal: (*float64)(&vt)})
-					case parser.DString:
-						row.Values = append(row.Values, driver.Datum{StringVal: (*string)(&vt)})
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_FloatVal{FloatVal: float64(vt)},
+						})
 					case parser.DBytes:
-						row.Values = append(row.Values, driver.Datum{BytesVal: []byte(vt)})
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_BytesVal{BytesVal: []byte(vt)},
+						})
+					case parser.DString:
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_StringVal{StringVal: string(vt)},
+						})
 					case parser.DDate:
-						row.Values = append(row.Values, driver.Datum{TimeVal: &driver.Datum_Timestamp{
-							Sec:  vt.Unix(),
-							Nsec: uint32(vt.Nanosecond()),
-						}})
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_TimeVal{
+								TimeVal: &driver.Datum_Timestamp{
+									Sec:  vt.Unix(),
```

Comment:
I don't think `Datum_Timestamp` needs to be a pointer here. Is there a `gogoproto` option to make it a value?